docs/DeveloperGuide.md

```diff
@@ -1,18 +1,616 @@
+that particular ```Ledger``` instance.
+
+For the sake of brevity, this section will focus on the discussion of the Manual Tracker.
+
+The Manual Tracker is capable of executing the following states of operation:
+
+|States| Operations | 
+|--------|----------|
+|```MAIN_MENU```|Go to main menu for users to choose the available operations
+|```CREATE_LEDGER```|Create a ledger specified by date, and append it to ```ledgerList```.
+|```DELETE_LEDGER```|Delete an existing ledger, referenced by date or index.
+|```OPEN_LEDGER```|Go to subroutine "Entry Tracker" for the entries recorded  under the specified ledger.
+
+#### Architecture
+
+##### Architecture Overview
+
+![](uml_images/manualTracker/images/Architecture_ManualTracker.png)
+
+
+|Module| Function | 
+|--------|----------|
+| ```Parser```|Parse inputs from user and return ```CommandPacket``` instance with organised ```commandString``` and ```paramMap``` 
+| ```Tracker/ Handler```|Manages the overall workflow of the Manual Tracker; identifies operation required from input and executes the corresponding ```command```.
+| ```Data``` |Refers to ```Ledger``` instances, stores relevant data of the day's transactions.
+| ```Data List``` |Refers to ```LedgerList``` instances, maintains Ledger instances within the program. 
+| ```Commands``` |Processes information from ```CommandPacket``` and executes the appropriate process from recognised params.
+| ```Logic``` |Outlines the abstract behavior of commands, as well as handle verification of params with appropriate error handling.
+
+
+
+##### Command and Logic
+
+![](uml_images/manualTracker/images/Commands_Logic_edited.png)
+
+|Class| Function |
+|--------|----------|
+|```retrieveLedgerCommand```| Process ```paramTypes```-```param``` pairs from the ```CommandPacket``` instance to identify specified ```Ledger``` instance, then retrieves the instance from the existing ```LedgerList```.
+|```createLedgerCommand```| Process ```paramTypes```-```param``` pairs from the ```CommandPacket``` instance to identify specified ```Ledger``` instance to be created, then creates the instance and append to existing ```LedgerList```.
+|```retrieveEntryCommand```| Omitted and left as exercise for reader. : ^ )
+|```createEntryCommand```| Omitted for brevity.
+|```editEntryCommand```| Omitted for brevity.
+|```ParamChecker```| Class contains a collection of methods that verify the correctness of the ```param``` supplied. <br><br> For instance, ```ParamChecker.checkAndReturnIndex``` checks if the index provided is out of bounds relative to the specified list, and throws the relevant exception if the input index is invalid. 
+|```ParamHandler```| Abstract class that outlines the general param handling behavior of ```commands``` instances and other classes that need to handle ```params``` in its operation.  
+
+##### Handler and Command
+
+![](uml_images/manualTracker/images/Handler_Commands.png)
```

Comment:
Is retriveEntryCommand, editEntryCOmmand, createEntryCommand properties of EntryTracker? Doesn't seem to be very clear based on the diagram as the table below doesn't really states how any of these EntryCommands are being used/ returned, etc. Also, is State a class or an Enumaration or? Would be better to make it clearer earlier